lib/unicorn/http_server.rb

```diff
@@ -385,13 +370,17 @@ def check_client_connection=(bool)
 
   # wait for a signal hander to wake us up and then consume the pipe
   def master_sleep(sec)
-    IO.select([ SELF_PIPE[0] ], nil, nil, sec) or return
-    SELF_PIPE[0].kgio_tryread(11)
+    @self_pipe[0].kgio_wait_readable(sec) or return
```

Comment:
https://github.com/betterplace/kgio/blob/master/ext/kgio/wait.c#L43

Doesn't need to select for a single fd - rb_wait_for_single_fd ( https://github.com/betterplace/kgio/blob/master/ext/kgio/wait.c#L11 ) :

```sh
Wed May  4 12:46:25 2011  KOSAKI Motohiro  <kosaki.motohiro@gmail.com>

	* thread.c (rb_wait_for_single_fd): Added POLLNVAL check.
	  based on a patch from Eric Wong at [ruby-core:35991].
```